vendor/bat-native-ledger/src/bat/ledger/internal/bat_helper.cc

```diff
@@ -2218,31 +2218,55 @@ std::string buildURL(const std::string& path,
   std::string url;
   switch (server) {
     case SERVER_TYPES::BALANCE:
-      if (ledger::is_production) {
-        url = BALANCE_PRODUCTION_SERVER;
-      } else {
-        url = BALANCE_STAGING_SERVER;
+      switch (ledger::_environment) {
```

Comment:
nit: worth considering decoupling to build the url from a simple factory, as this way we just need to pass in the type of url, i.e. balance

Reply:
can you please log an issue for it so that we don't forget